debug/org.eclipse.debug.ui/ui/org/eclipse/debug/internal/ui/views/memory/renderings/RenderingsUtil.java

```diff
@@ -557,21 +557,21 @@ public static byte[] convertHexStringToByteArray(String str, int numBytes, int n
 			// convert string to byte
 			String oneByte = str.substring(i*2, i*2+2);
 
-			Integer number = Integer.valueOf(oneByte, 16);
-			if (number.compareTo(Integer.valueOf(Byte.toString(Byte.MAX_VALUE))) > 0)
+			int number = Integer.parseInt(oneByte, 16);
+			if (Integer.compare(number, Integer.valueOf(Byte.toString(Byte.MAX_VALUE))) > 0)
 			{
-				int temp = number.intValue();
+				int temp = number;
 				temp = temp - 256;
 
 				String tempStr = Integer.toString(temp);
 
-				Byte myByte = Byte.valueOf(tempStr);
-				bytes[i] = myByte.byteValue();
+				byte myByte = Byte.parseByte(tempStr);
+				bytes[i] = myByte;
 			}
 			else
 			{
-				Byte myByte = Byte.valueOf(oneByte, 16);
-				bytes[i] = myByte.byteValue();
+				byte myByte = Byte.parseByte(oneByte, 16);
+				bytes[i] = myByte;
 			}
 		}
```

Comment:
Unrelated to the application of the automated cleanup: this looks like a very round-about way to convert a hex string to a byte. What's wrong with `bytes[i] = Integer.parseInt(oneByte, 16).byteValue()`?

Reply:
> this looks like a very round-about way to convert a hex string to a byte

Yes, that's what this code seems to be about: it converts a substring containing a single byte representation to a byte. But I think I do not understand your proposal yet. What would be the benefit of replacing `Byte.parseByte(oneByte, 16)` with `Integer.parseInt(oneByte, 16).byteValue()`?

Reply:
> What's wrong with bytes[i] = Integer.parseInt(oneByte, 16).byteValue()

@tomaswolf  `Integer.parseInt` returns an `int` (primitive) so `Integer.parseInt(oneByte, 16).byteValue()` doesn`t even compile :-\

Reply:
> `Integer.parseInt` returns an `int` (primitive) so `Integer.parseInt(oneByte, 16).byteValue()` doesn`t even compile :-\

You can box the returned `int` into an `Integer`.

Reply:
> > `Integer.parseInt` returns an `int` (primitive) so `Integer.parseInt(oneByte, 16).byteValue()` doesn`t even compile :-\
> 
> You can box the returned `int` into an `Integer`.

That would by an overkill (parsing + boxing) and would also skip the range validation since `Integer::byteValue` simply casts to `byte` while `Byte::parseByte` does validate the range (and throw a `NumberFormatException` if the parsed byze is out of range).

I'd go with the original form: `Byte.parseByte(oneByte, 16)`

Reply:
I meant the whole code looks way too complicated. I think one can replace all of this from line 560 to line 575 with just
`bytes[i] = (byte) Integer.parseInt(oneByte, 16).intValue();`.

Reply:
> would also skip the range validation since `Integer::byteValue` simply casts to `byte` while `Byte::parseByte` does validate the range

The input already is a byte value, which is why the range check is not necessary.

> I meant the whole code looks way too complicated. I think one can replace all of this from line 560 to line 575

You are right. I first thought it would make a difference depending on whether the string to parse contains a signed or unsigned value, but that's not the case. So your proposal totally makes sense to me. So I would suggest to provide a PR to simplify this code. Do want to provide a PR or shall I, @tomaswolf?

Reply:
Go ahead; I currently have no development setup for Eclipse Platform and am too busy in other areas. Just noticed this by chance.

Reply:
Thank you, @tomaswolf! I have created #1316 with the simplification. Could you please check if that properly reflects your proposal? I will not further process this PR until the other is merged, so that rebasing this on top will removing the cleanup for the code part that we have discussed.